test/mongodb.test.js

```diff
@@ -1428,10 +1428,184 @@ describe('mongodb connector', function() {
+              },
+            },
+          }, function(err, results) {
+            should.not.exist(err);
+            should.exist(results);
+            results.length.should.be.equal(3);
+            var dist = 0;
+            results.forEach(function(result) {
+              var currentDist = testUtils.getDistanceBetweenPoints(coords, result.location);
+              currentDist.should.be.aboveOrEqual(dist);
+              currentDist.should.be.belowOrEqual(17);
               dist = currentDist;
             });
+            done();
+          });
+        });
+      });
+    });
+
+    it('find should be able to query by location via near with minDistance set', function(done) {
+      var coords = { lat: 30.274085, lng: 120.15507000000002 };
+      geoDb.autoupdate(function(err) {
+        var createPost = function(far) {
+          var point;
+          if (far) {
+            point = new GeoPoint({
+              lat: 31.230416,
+              lng: 121.473701,
+            });
+          } else {
+            point = new GeoPoint({
+              lat: 30.27167 + Math.random() * 0.01,
+              lng: 120.13469600000008 + Math.random() * 0.01,
+            });
+          }
+          return function(callback) {
+            PostWithLocation.create({ location: point }, callback);
+          };
+        };
 
+        async.parallel([
+          createPost(false),
+          createPost(false),
+          createPost(false),
+          createPost(true),
+        ], function(err) {
+          should.not.exist(err);
+
+          PostWithLocation.find({
+            where: {
+              location: {
+                near: new GeoPoint(coords),
+                minDistance: 17000,
+                unit: 'meters',
+              },
+            },
+          }, function(err, results) {
+            should.not.exist(err);
+            should.exist(results);
+            var dist = 0;
+            results.forEach(function(result) {
+              var currentDist = testUtils.getDistanceBetweenPoints(coords, result.location);
+              currentDist.should.be.aboveOrEqual(dist);
+              dist = currentDist;
+            });
+
+            done();
+          });
+        });
+      });
+    });
+
+    it('find should be able to set unit when  query location via near', function(done) {
+      var coords = { lat: 30.274085, lng: 120.15507000000002 };
+
+      geoDb.autoupdate(function(err) {
+        var createPost = function(far) {
+          var point;
+          if (far) {
+            point = new GeoPoint({
+              lat: 31.230416,
+              lng: 121.473701,
+            });
+          } else {
+            point = new GeoPoint({
+              lat: 30.27167 + Math.random() * 0.01,
+              lng: 120.13469600000008 + Math.random() * 0.01,
+            });
+          }
+          return function(callback) {
+            PostWithLocation.create({ location: point }, callback);
+          };
+        };
```

Comment:
Please extract createPost into a shared helper function, I noticed you also used it above at line 1498